exams/utils.py

```diff
@@ -59,3 +67,53 @@ def authorize_for_exam(mmtrack, course_run):
                 mmtrack.user.username,
                 course_run.edx_course_key
             )
+        else:
+            log.info(
+                'Unable to authorize user "%s" for exam, course id "%s"',
+                mmtrack.user.username,
+                course_run.edx_course_key
+            )
+
+
+def bulk_authorize_for_exam(program_id=None, username=None, course_key=None):
+    """
+    Authorize user(s) for exam(s).
+
+    Args:
+        program_id(str): Program id (optional) if you want authorization on spcific program exams
+        username(str): User name (optional) whom you want to authorize for exam(s)
+        course_key(str): Edx course key  (optional) where you want to authorize user(s)
+    """
+    programs = Program.objects.filter(exam_series_code__isnull=False, live=True).filter(~Q(exam_series_code=""))
+    if program_id:
+        programs = programs.filter(id=program_id)
+
+    if not programs.exists():
+        log.info('[authorization_user_exam] Program(s) are not available for exam authorization.')
+        return
+
+    for program in programs:
+        program_enrollment_qset = ProgramEnrollment.objects.filter(program=program)
+        if username:
+            program_enrollment_qset = program_enrollment_qset.filter(user__username=username)
+
+        if not program_enrollment_qset.exists():
+            log.info('[authorization_user_exam] User(s) not exist for exam authorization.')
+            return
+
+        for program_enrollment in program_enrollment_qset:
+            user = program_enrollment.user
+            enrollments_qset = CachedEnrollment.user_course_qset(user, program=program)
+            if course_key:
+                enrollments_qset = enrollments_qset.filter(course_run__edx_course_key=course_key)
+
+            if not enrollments_qset.exists():
+                log.info('[authorization_user_exam] Course(s) not exist for exam authorization.')
+                return
+
+            mmtrack = get_mmtrack(user, program)
+
+            for enrollment in enrollments_qset:
+                # if user has passed and paid for the course
+                # and not already authorized for exam the create authorizations.
+                authorize_for_exam(mmtrack, enrollment.course_run)
```

Comment:
I'm not sure we want to authorize the user for *every* course_run the user ever took. @pdpinch I know you mentioned we want to check all the course runs, but do we want to do authorize for all runs or do something more targeted like only authorize for the most recent course run the user is eligible to take an exam for?